enumerable.rb

```diff
@@ -0,0 +1,128 @@
+# rubocop:disable Metrics/CyclomaticComplexity
+# rubocop:disable Metrics/PerceivedComplexity
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    array = to_a
+    array.length.times { |element| yield(array[element]) }
+    self
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    array = to_a
+    array.length.times { |element| yield(array[element], element) }
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    new_arr = []
+    my_each { |it| new_arr << it if yield(it) }
+    new_arr
+  end
+
+  def my_all?(*arg)
+    return "`my_all?': wrong # of arguments (given #{arguments.length}, expected 0..1)" if arguments.length > 1
+
+    if block_given?
+      my_each { |element| return false unless yield(element) }
+    elsif arg[0].is_a? Class
+      my_each { |element| return false unless element.class.ancestors.include?(arg[0]) }
+    elsif arg[0].is_a? Regexp
+      my_each { |element| return false unless arg[0].match(element) }
+    elsif arg.empty?
+      return include?(nil) || include?(false) ? false : true
+    else
+      my_each { |element| return false unless element == arg[0] }
+    end
+    true
+  end
```

Comment:
- [ ] `my_all` is not working as the original `all?` method, also is not passing all test cases. 

### `all?` is an Enumerable that has these characteristics:
- Returns `true` if all elements are not false or nil.
```
[1, 'string', true].all?
# true
```
- Returns `false` if it finds at least one element that is not true or nil
```
[1, 'string', nil].all?
# false
```
- it doesn't change the original array/range/hash
- it can accept as arguments a Class, a RegEx, or a pattern, and returns true or false depending if the values in the array/range/hash match the condition
- it can use a block but it is not necessary to make the method work

[Enumerable#all?](https://ruby-doc.org/core-2.7.2/Enumerable.html#method-i-all-3F)